### PR TITLE
Main UI 

### LINKED ISSUE #30: Email sender - main page UI
Finish the email sender page UI in `pages/emailSender.tsx`. It should look and behave like the wireframes in the figma here https://www.figma.com/file/BQTi7tgxKpd54q3MaeAcbN/Email-sender-wireframe. Add the "Send" button, and add a way to display either a success message or an error message. The functions to actually send the email will be developed in parallel so for the purposes of this ticket you can just display the success/error messages after a user clicks a button.

### PR DESCRIPTION
**Pull request by mike10911**
Closes #30
This PR includes the following changes:

- Added a send button to ‘send’ emails
- When send is clicked, currently we’re mocking that even number indexed messages will error
- After send, if there are errors, each message with an error will include an error message
- After send, displays a red colored result message with num failed / total if there are errors
- After send, displays a mint colored result message if there are no errors

Note : If you want to test that all messages are successfully sent, in the sendEmails function, swap out i % 2 === 0 for false

https://user-images.githubusercontent.com/103906123/176821752-a222c60f-716b-43e8-9120-50474ab25325.mp4

Screen Shot 2022-06-30 at 11 53 44 PM
Screen Shot 2022-06-30 at 11 53 10 PM

### CODE REVIEW

**Review by karynayen — CHANGES_REQUESTED** [review]
Wow this looks BEAUTIFUL!!! The only think I might suggest would be to move the error/success messages right under the send emails button so that users don’t have to scroll all the way to the bottom of the page to see the results?

File: pages/emailSender.tsx [hunk 1]
```diff
@@ -165,95 +188,110 @@ const EmailSender: NextPage = () => {
...
+                  accept={'.csv'}
+                  type="file"
+                  onChange={handleOnChange}
+                />
+                <label htmlFor="contained-button-file">
+                  <Button variant="contained" component="span">
+                    Upload
+                  </Button>
+                </label>
+                <StyledCsvButton
```
Comment (deanframe) [inline pages/emailSender.tsx:224]: We should disable this button visually (gray it our or something) if no csv has been uploaded yet

File: pages/emailSender.tsx [hunk 2]
```diff
@@ -118,6 +122,16 @@ const EmailSender: NextPage = () => {
...
   }
 
+  const sendEmails = () => {
+    const localErrorMessages : ErrorMessage[] = []
+    for (let i = 0; i < finalMessages.length; i++) {
+      if (i % 2 === 0) {
+        localErrorMessages.push({ id: finalMessages[i].id, message: 'Errorrrr!' })
+      }
+    }
+    setResultErrorMessage({ errorMessages: localErrorMessages, resultMessage: { isError: localErrorMessages.length > 0, message: localErrorMessages.length > 0 ? `Error sending ${localErrorMessages.length} of ${finalMessages.length}` : 'Sent emails successfully' } })
```
Comment (deanframe) [inline pages/emailSender.tsx:132]: Nit: hmm this line's a little long - does the eslint hook check for that?

File: pages/emailSender.tsx [hunk 3]
```diff
@@ -165,95 +188,110 @@ const EmailSender: NextPage = () => {
...
+                  }}
+                >
+                  Import CSV!
+                </StyledCsvButton>
+              </StyledCsvButtonsContainer>
+            </SectionContainer>
+          </FormControl>
+          <StyledTableContainer>
+            <TableContainer component={Paper}>
+              <StyledTable aria-label="simple table">
```
Comment (deanframe) [inline pages/emailSender.tsx:238]: Nit: we can probably come up with a better aria-label here for accessibility

File: pages/emailSender.tsx [hunk 4]
```diff
@@ -165,95 +188,110 @@ const EmailSender: NextPage = () => {
...
+              width="medium"
+            >
+              Print final messages
+            </StyledButton>
+            </SectionContainer>
+            <br />
+            <br />
+            <SectionContainer>
+            <StyledSubHeader variant="h5">4) Send emails</StyledSubHeader>
+            <StyledButton
```
Comment (deanframe) [inline pages/emailSender.tsx:279]: Same as above- let's hide/disable the buttons on the page until they are actionable

**Review by deanframe — CHANGES_REQUESTED** [review]
Looks mostly great! Let's change buttons to be grayed out/disabled until they'll actually do something when pressed.

**Comment by deanframe** [comment]
Also, shouldn't `yarn.lock` be gitignored?

File: pages/emailSender.tsx (hunk 2 above)
Comment (jzhzhang66) [reply pages/emailSender.tsx:132]: No I don't think it does - just added!

**Comment by jzhzhang66** [comment]
@deanframe that's a good point - I wasn't sure either and just googled it but it seems like people are saying we should include it? 
Screen Shot 2022-07-01 at 5 00 15 PM
https://classic.yarnpkg.com/blog/2016/11/24/lockfiles-for-all/

**Review by karynayen — APPROVED** [review]
AMAZING

**Review by deanframe — APPROVED** [review]
Woohoo!!!

**Pull request merged**